README.md
# ReNaze Technologies
Привет! Ты попал на страничку свежего проекта производства компании ReNaze,
который позволит вам удобно и автоматически фармить звезды на /quake

Проект в разработке, будем рады любой поддержке.

---

[**Mini-documentation**](something/docs.md)

[**Changelog** ***0.2v***](something/changelog.md)

[**Our VK-group**](https://vk.com/renaze)

something/changelog.md
## Changelog
> 0.2v
* Добавлен changelog.md (чё)
* Завезена удобная менюшка при открытии батника (относительно удобная)
* Переделаны .js файлы, теперь все классно наверно
* Добавлена поддержка проху (проху хуета не юзайте) (ладно наебал мне лень делать прокси)
* Если на /quake посторонние игроки, боты покидают игру и анрегаются
* Автоматический анрег прошлых аккаунтов

> 0.1v (релиз)
* Простая автоматическая системка фарма ботами на /quake
* Ручной ввод количества игр, которые сыграют боты
* Анрег ботов в случае помехи

---

[**Our VK-group**](https://vk.com/renaze)

app.js
const mf = require('mineflayer'); // Коренной модуль к MineCraft
const fs = require('fs');

let nicks, proxy;
fs.readFile("./something/nick.txt", 'utf8', function(e,d) { 
 if(e) throw e; nicks = d.replace(/\r/g,'').split('\n');
 });
console.log(nicks)

const nickname = process.argv[2];
var tries = Number(process.argv[3]);
if(tries && tries > 0 && tries < 30) tries = Number(tries);
else tries = 1;

let worker_win = false;

async function main() {
 await sleep(100);
 let workers = [{},{},{}]; let i = 0;
 while(i < 3) {
  let worker = nicks[utils.random(0,nicks.length-1)];
  worker = worker.split(''); worker[utils.random(0,worker.length-1)] = '' + utils.random(0,9) + '';
  console.log(worker.join(''))
  workers[i] = { nick: worker.join(''), login: false };
  i += 1;
 };
 
 
	await sleep(100);
	//console.log(nicks)
 
 //workers = [{nick:"hello",login:false},{nick:"byebye",login:false},{nick:"suckmyfoxecock",login:false}];
 
 for await(let worker of workers) {
 let bot = mf.createBot({
  username: worker.nick,
  host: 'minecorp.ru', port: 25541,
  version: '1.12.2'
 });
 let err;
 
 bot.on('login', async() => {
  await sleep(200);
  bot.chat(`/reg Lalala Lalala`); bot.chat(`/l Lalala`);
  await sleep(50);
  bot.setControlState('back', true);
  await sleep(utils.random(300,500));
  worker.login = true; worker.bot = bot;
 });
 
 bot.on('message', (msg) => {
	console.log(`[${worker.nick}] ` + msg.toAnsi()); 
 });
 
 await sleep(2400);
 if(!worker.login) {
  console.log(`${worker.nick} не удалось войти`);
  bot = null; break;
 };
 bot.clearControlStates();
 };
 
 if(workers.find(x=> !x.login)) return console.log(`(!) Не всем ботам удалось войти`);
 //console.log(workers)
 var works = 0;
 
 workers[1].bot.on('message', async function(msg) {
 let part, winner;
 part = msg.toString().split(' ');
 if(part[1] === 'победил' && part[3] === '/quake') {
	winner = workers.find(x=> x.nick === part[0]);
	if(winner) {
		worker_win = true;
		await sleep(3000);
		winner.bot.chat(`/pay ${nickname} 0.3`);
		
		//workers[0].bot.chat(`/server creative`);
 
       await sleep(5000);
       works += 1; work(workers);
		
	}
 };
 });
 
 
 async function work(workers) {
 worker_win = false;
 if(works >= tries) {
 //workers[0].bot.chat(`/pay ${nickname} ${tries*0.3}`);
 await sleep(100);
 
 end(workers);
 console.log("Всё! :)")
 return;
 };
 
 for await(let worker of workers) {
  worker.bot.chat(`/quake`); await sleep(100);
 };
 await sleep(26000);
 /*let tab = Object.keys(workers[0].bot.players);
 console.log(tab);
 if(tab.find(x=> !x.match('§') && !workers.find(y=> y.nick === x))) {
	 console.log(tab.find(x=> !x.match('§') && !workers.find(y=> y.nick === x)));
 workers[0].bot.chat('Упс, пока.');
 await sleep(50);
 for await(let worker of workers) {
  worker.bot.chat(`/hub`);
  await sleep(200);
  worker.bot.chat(`/unregister Lalala`);
  await sleep(50);
  worker.bot.quit();
 };
 }; // Посторонний игрок*/ /// Нерабочая функция!!!
 await sleep(14000);
 workers[1].bot.chat(`/server creative`);
 workers[2].bot.chat(`/server creative`);
 await sleep(15000);
 if(!worker_win) {
  workers[0].bot.chat(`/server creative`);
  await sleep(100);
  end(workers);
  console.log(`(!) Похоже, что ботам помешали`);
 };
 };
 
 work(workers);
};

async function end(workers) {
 for await(let worker of workers) {
   worker.bot.chat(`/unregister Lalala`);
   await sleep(100);
   worker.bot.quit(); worker.bot.end();
  };
 console.log(`> Остановлено, для запуска снова перезапустите консоль`);
};


const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
const utils = {
	random: (x, y) => {
	return Math.floor(x + Math.random() * (y + 1 - x));
    }
};


main();
// Фармилка звёзд MineCorp Creative (☆)

main.js
const readline = require("readline"); 
const config = require('./config');

console.clear();
console.log("\x1b[36m", '> MineCorp Creative Farmer')
console.log("\x1b[37m", '                      v0.2')

function end(count) {
 if(count >= 2 && count <= 4) return 'а';
 return '';
}; // Окончание в слове 'раз'

async function main() {
const r1 = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const { promisify } = require('util');
r1.question[promisify.custom] = (question) => {
  return new Promise((resolve) => {
    r1.question(question, resolve);
  });
};

console.log('');

if(!config.nickname) {
	config.nickname = await promisify(r1.question)('Укажите свой никнейм: ');
	require('fs').writeFileSync('./config.json', JSON.stringify(config, null, '\t'));
} else console.log(`! Никнейм взят из конфига`);

//let times = await promisify(r1.question)('> Укажите число игр, которые сыграют боты: ');

let farm_started = false;
let times = 3;

console.log(`* Установки:
  Никнейм: ${config.nickname} (изменить: "nick JuniorFrr")
  Боты играют: ${times} раз${end(times)} (изменить: "times 1")`);
console.log('');

async function command() {
 let cmd = (await promisify(r1.question)('> ')).split(' ');
 
 let caps = cmd[0];
 if(caps.toUpperCase() === caps) {
 console.log(`! НЕ КАПСИ НА МЕНЯ ДЕБИЛ`);
 return command();
 };
 
 
 
 if(cmd[0] === 'start') {
  if(farm_started) {
  console.log(`(!) Фармилка уже запущена`);
  return command(); 
  };
  require('./utils/quake_farmer.js')(times, farm_started, command);
  console.log(`Фармилка запущена`);
  return;
 };
 
 if(cmd[0] === 'times' && Number(cmd[1])) {
  if(Number(cmd[1]) < 1 || Number(cmd[1]) > 30 || Number(cmd[1]).toFixed() != Number(cmd[1])) {
   console.log(`Укажите число в районе от 1 до 30`);
   return command();
  };
  times = Number(cmd[1]);
  console.log(`Теперь боты сыграют /quake: ${times} раз${end(times)}`);
  return command();
 };
 
 if(cmd[0] === 'nick' && cmd[1]) {
  config.nickname = cmd[1];
  console.log(`Ваш никнейм изменен на ${cmd[1]}`);
  return command();
 };
 
 if(cmd[0] === 'unreg' && cmd[1]) {
  require("./utils/unreg.js")(cmd.join(' ').replace('unreg ',''), command);
  return;
 };
 
 if(cmd[0] === 'meow') {
  console.log(`Мяу-мяу.`);
  return command();
 };
 
 if(cmd[0] === 'фыр') {
  console.log(`Уру-ру.`);
  return command();
 };
 
 if(cmd[0] === 'иди' && cmd[1] === 'нахуй') {
  console.log(`ок`);
  return command();
 };
 
 
 console.log(cmd)
 if(cmd[0] === '') return command();
 console.log(`Команда не опознана (${cmd.join(' ')})`);
 command();
};
command();

r1.on("close", function() {
    console.log("\n> Пока!");
    process.exit(0);
});

}; // Консоль
main();



utils/unreg.js
// Анрегистрация ботов (ручная)

module.exports = async function(string) {
const mf = require('mineflayer');
let unreg = string.split(',');
console.log(unreg)

const config = require('../config.json');

for(let player of unreg) {
player = player.replace(/[^\wа-я_]/gi,'');
console.log(`> Unregistering ${player}`);
if(config.unregister.find(x=> x === player)) config.unregister.splice(config.unregister.indexOf(player), 1);

let bot = mf.createBot({
 username: player,
 host: 'hub.minecorp.ru',
 version: '1.12.2'
});

bot.on('login', function() {
 setTimeout(() => { bot.chat(`/login Lalala`) }, 500);
 setTimeout(() => { bot.chat(`/unregister Lalala`) }, 1000);
 setTimeout(() => { bot.end() }, 1500);
});

bot.on('message', function(msg) {
 console.log(`[${player}] ` + msg.toAnsi());
});
};

console.log('Meow')
console.log(process.argv)
//console.log(bot)
setTimeout(() => { command() }, 5000);
};

something/prev_main.js
const mf = require('mineflayer'); // Коренной модуль к MineCraft
const ProxyAgent = require('proxy-agent'); // Модуль для прокси
const fs = require('fs');

const config = require('./config'); // Конфиг

let nicks, proxy;
fs.readFile("./something/nick.txt", 'utf8', function(e,d) { 
 if(e) throw e; nicks = d.replace(/\r/g,'').split('\n');
 }); // Список используемых никнеймов

if(config.proxy) {
fs.readFile("./something/proxy.txt", 'utf8', function(e,d) { 
 if(e) return fs.writeFileSync('./something/proxy.json', []);
 proxy = d.replace(/\r/g,'').split('\n');
 for(let ip of proxy) { ip = ip.split(':'); ip[1] = Number(ip[1]) };
 });
}; // Список прокси







async function main() {
 await sleep(100);
 let workers = [{},{},{}]; let i = 0;
 while(i < 3) {
  let worker = nicks[utils.random(0,nicks.length-1)];
  worker = worker.split(''); worker[utils.random(0,worker.length-1)] = '' + utils.random(0,9) + '';
  console.log(worker.join(''))
  workers[i] = { nick: worker.join(''), login: false };
  i += 1;
 };
 
 await sleep(100);
 
 
 for await(let worker of workers) {
 let bot = mf.createBot({
  username: worker.nick,
  host: 'minecorp.ru', port: 25541,
  version: '1.12.2'
 });
 let err;
 
 bot.on('login', async() => {
  await sleep(200);
  bot.chat(`/reg Lalala Lalala`); bot.chat(`/l Lalala`);
  await sleep(50);
  bot.setControlState('back', true);
  await sleep(utils.random(300,500));
  worker.login = true; worker.bot = bot;
 });
 
 bot.on('message', (msg) => {
	console.log(`[${worker.nick}] ` + msg.toAnsi()); 
 });
 
 await sleep(2400);
 if(!worker.login) {
  console.log(`${worker.nick} не удалось войти`);
  bot = null; break;
 };
 bot.clearControlStates();
 };
 
 if(workers.find(x=> !x.login)) return console.log(`(!) Не всем ботам удалось войти`);
 //console.log(workers)
 var works = 0;
 
 workers[1].bot.on('message', async function(msg) {
 let part, winner;
 part = msg.toString().split(' ');
 if(part[1] === 'победил' && part[3] === '/quake') {
	winner = workers.find(x=> x.nick === part[0]);
	if(winner) {
		worker_win = true;
		await sleep(3000);
		winner.bot.chat(`/pay ${nickname} 0.3`);
		
		//workers[0].bot.chat(`/server creative`);
 
       await sleep(5000);
       works += 1; work(workers);
		
	}
 };
 });
 
 
 async function work(workers) {
 worker_win = false;
 if(works >= tries) {
 //workers[0].bot.chat(`/pay ${nickname} ${tries*0.3}`);
 await sleep(100);
 
 end(workers);
 console.log("Всё! :)")
 return;
 };
 
 for await(let worker of workers) {
  worker.bot.chat(`/quake`); await sleep(100);
 };
 await sleep(26000);
 await sleep(14000);
 workers[1].bot.chat(`/server creative`);
 workers[2].bot.chat(`/server creative`);
 await sleep(20000);
 if(!worker_win) {
  workers[0].bot.chat(`/server creative`);
  await sleep(100);
  end(workers);
  console.log(`(!) Похоже, что ботам помешали`);
 };
 };
 
 work(workers);
};

async function end(workers) {
 for await(let worker of workers) {
   worker.bot.chat(`/unregister Lalala`);
   await sleep(100);
   worker.bot.quit(); worker.bot.end();
  };
 console.log(`> Остановлено`);
};


const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
const utils = {
	random: (x, y) => {
	return Math.floor(x + Math.random() * (y + 1 - x));
    }
};


main();
// Фармилка звёзд MineCorp Creative (☆)

utils/quake_farmer.js
// Фармилка звёзд MineCorp Creative
//
// quake_farmer.js

module.exports = async function(tries, farm_started) {
const mf = require('mineflayer'); // Коренной модуль к MineCraft
const ProxyAgent = require('proxy-agent'); // Модуль для прокси
const fs = require('fs');

const config = require('../config'); // Конфиг

let nicks, proxy;
fs.readFile("./something/nicks.txt", 'utf8', function(e,d) { 
 if(e) throw e; nicks = d.replace(/\r/g,'').split('\n');
 }); // Список используемых никнеймов

if(config.proxy) {
fs.readFile("./something/proxy.txt", 'utf8', function(e,d) { 
 if(e) return fs.writeFileSync('./something/proxy.json', []);
 proxy = d.replace(/\r/g,'').split('\n');
 for(let ip of proxy) { ip = ip.split(':'); ip[1] = Number(ip[1]) };
 });
}; // Список прокси





async function main() {
 await sleep(200);
 
 if(config.unregister.length) {
 for(let worker of config.unregister) {
 let bot = mf.createBot({
  username: worker,
  host: 'creative.minecorp.ru', port: 25541,
  version: '1.12.2'
 });
 
 console.log(`! Unregistering ${worker}...`)
 bot.on('message', (msg) => { console.log(msg.toAnsi()) });
 
 bot.once('login', async() => {
  bot.chat(`/l Lalala`);
  await sleep(20);
  bot.setControlState('back', true);
  await sleep(1500);
  bot.chat(`/unreg Lalala`);
  await sleep(100);
  bot.quit(); bot.end();
  console.log(`! ${worker} unregistered`)
 });
 bot.once('kicked', async(reason) => { console.log(reason) });
 
 await sleep(3000);
 };
 await sleep(2000);
 config.unregister = [];
 };
 
 let workers = [{},{},{}]; let i = 0;
 while(i < 3) {
  let worker = nicks[utils.random(0,nicks.length-1)];
  if(config.shuff_digits) {
  worker = worker.split(''); worker[utils.random(0,worker.length-1)] = '' + utils.random(0,9) + '';
  console.log(worker.join(''))
  worker = worker.join('');
  };
  if(workers.find(x=> x.nick === worker)) continue;
  workers[i] = { nick: worker, login: false };
  i += 1; config.unregister.push(worker);
 };
 
 require('fs').writeFileSync('./config.json', JSON.stringify(config, null, '\t'));
 await sleep(100);
 
 for await(let worker of workers) {
 let bot = mf.createBot({
  username: worker.nick,
  host: 'minecorp.ru', port: 25541,
  version: '1.12.2'
 });
 let err;
 worker.bot = bot;
 
 bot.once('login', async() => {
  await sleep(200);
  bot.chat(`/reg Lalala Lalala`); bot.chat(`/l Lalala`);
  config.unregister.push(worker.nick);
  await sleep(50);
  bot.setControlState('back', true);
 });
 
 bot.on('message', (msg) => {
	console.log(`[${worker.nick}] ` + msg.toAnsi());
	if(msg.toString() === 'Вы успешно вошли!') worker.login = true;
 });
 
 bot.once('end', () => { worker.login = false; });
 bot.once('kicked', (reason) => { console.log(reason) });
 
 await sleep(5000);
 if(!worker.login) {
  console.log(`! ${worker.nick} не удалось войти`);
  bot = null; break;
 };
 bot.clearControlStates();
 };
 
 if(workers.find(x=> !x.login)) {
	 end(workers);
	 console.log(`! Не всем ботам удалось войти`);
	 return;
 };
 //console.log(workers)
 var works = 0;
 
 workers[1].bot.on('message', async function(msg) {
 let part, winner;
 part = msg.toString().split(' ');
 if(part[1] === 'победил' && part[3] === '/quake') {
	winner = workers.find(x=> x.nick === part[0]);
	if(winner) {
		worker_win = true;
		workers[0].bot.setControlState('back', true);
		await sleep(1000);
		winner.bot.chat(`/pay ${config.nickname} 0.3`);
		
		//workers[0].bot.chat(`/server creative`);
 
       await sleep(3000);
	   workers[0].bot.clearControlStates();
       works += 1; work(workers);
		
	}
 };
 });
 
 
 async function work(workers) {
 if(works >= tries) {
 //workers[0].bot.chat(`/pay ${nickname} ${tries*0.3}`);
 await sleep(100);
 
 end(workers);
 console.log("! Всё! :)")
 return;
 };
 
 for await(let worker of workers) {
  worker.bot.chat(`/quake`); await sleep(100);
 };
 
 await sleep(26000);
 await sleep(14000);
 workers[1].bot.chat(`/server creative`);
 workers[2].bot.chat(`/server creative`);
 
 let anyone = [];
 
 console.log(workers[0].bot.scoreboard)
 console.log('ALL SC ^^^^')
 console.log(workers[0].bot.scoreboard[1])
 console.log(workers[0].bot.scoreboard[1].itemsMap)
 console.log(workers[0].bot.scoreboard[1][0])
 
 
 for(let player of Object.keys(workers[0].bot.scoreboard[1].itemsMap)) {
 if(workers.find(x=> x.nick === player.replace(/[^\w_а-я]/gi,''))) continue;
 anyone.push(player);
 };
 await sleep(200);
 if(anyone.length) {
  workers[0].bot.chat(`Упс, пока.`);
  await sleep(100);
  workers[0].bot.chat(`/server creative`);
  await sleep(100);
  end(workers);
  console.log(`! Похоже, что ботам помешали (${anyone.join(', ')})`);
 };
 };
 
 work(workers);
};

async function end(workers) {
 for await(let worker of workers) {
   if(worker.bot) worker.bot.chat(`/unregister Lalala`);
   await sleep(100);
   if(worker.bot) { worker.bot.quit(); worker.bot.end() };
  };
 config.unregister = [];
 require('fs').writeFileSync('./config.json', JSON.stringify(config, null, '\t'));
 await sleep(200);
 console.log(`! Фармилка /quake остановлена`);
 return command();
};


const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
const utils = {
	random: (x, y) => {
	return Math.floor(x + Math.random() * (y + 1 - x));
    }
};


main();
};